libpince/utils.py

```diff
@@ -767,18 +767,10 @@ def aob_to_str(list_of_bytes, encoding="ascii"):
                 byte = sByte
             else:
                 byte = int(sByte, 16)
-            """NOTE: replacing non-printable chars with a period will
-            have an adverse effect on the ability to edit hex/ASCII data
-            since the editor dialog will replace the hex bytes with 2e rather
-            than replacing only the edited bytes.
-
-            So for now, don't replace them -- but be aware that this clutters
-            the ascii text in the memory view and does not look 'neat'
-            """
-            # if ( (byte < 32) or (byte > 126) ):
-            #    hexString += f'{46:02x}' # replace non-printable chars with a period (.)
-            # else:
-            hexString += f"{byte:02x}"
+            if (byte < 32) or (byte > 126):
```

Comment:
HexEditDialog (right click -> edit) will convert all data to literal dots for representation. This will cause hex data `2E` to be written to all garbage bytes upon changing the ascii representation. The solution could be to preserve the old behavior for HexEditDialog. Any other ideas?

Reply:
added a "replace_unprintable=True" parameter. This will allow to decide per use case.
- replace_unprintable set to false for the EditHexDialog

Reply:
Yeah, that makes sense, will be waiting for your commit